hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticClient.java

```diff
@@ -0,0 +1,474 @@
+@ConcurrencyManagement(ConcurrencyManagementType.BEAN)
+public class ElasticClient {
+  private static final Logger LOG = Logger.getLogger(ElasticClient.class.getName());
+  
+  @EJB
+  private Settings settings;
+  private Client elasticClient = null;
+  
+  @PostConstruct
+  private void initClient() {
+    try {
+      getClient();
+    } catch (ServiceException ex) {
+      LOG.log(Level.SEVERE, "", ex);
+    }
+  }
+  @PreDestroy
+  private void closeClient(){
+    shutdownClient();
+  }
+  
+  /**
+   * Shuts down the client
+   * <p/>
+   */
+  private void shutdownClient() {
+    if (elasticClient != null) {
+      elasticClient.close();
+      elasticClient = null;
+    }
+  }
+  
+  private synchronized Client getClient() throws ServiceException {
+    if (elasticClient == null) {
+      LOG.log(Level.INFO, "creating new elastic client");
+      final org.elasticsearch.common.settings.Settings settings
+        = org.elasticsearch.common.settings.Settings.builder()
+        .put("client.transport.sniff", true) //being able to retrieve other nodes
+        .put("cluster.name", "hops").build();
+      
+      List<String> elasticAddrs = getElasticIpsAsString();
+      TransportClient _client = new PreBuiltTransportClient(settings);
+      for(String addr : elasticAddrs){
+        _client.addTransportAddress(new TransportAddress(new InetSocketAddress(addr, this.settings.getElasticPort())));
+      }
+      elasticClient = _client;
+      
+      Iterator<ThreadPool.Info> tpInfoIt = elasticClient.threadPool().info().iterator();
+      StringBuilder tp = new StringBuilder();
+      while(tpInfoIt.hasNext()) {
+        ThreadPool.Info tpInfo = tpInfoIt.next();
+        switch(tpInfo.getName()) {
+          case ThreadPool.Names.BULK:
+          case ThreadPool.Names.INDEX:
+          case ThreadPool.Names.SEARCH:
+            tp.append("name:").append(tpInfo.getName())
+              .append("(")
+              .append(tpInfo.getThreadPoolType())
+              .append(",")
+              .append(tpInfo.getMin())
+              .append(",")
+              .append(tpInfo.getMax())
+              .append(",")
+              .append(tpInfo.getQueueSize().singles())
+              .append(")");
+        }
+      }
+      LOG.log(Level.INFO, "threadpools {0}", tp);
+    }
+    return elasticClient;
+  }
+  
+  private List<String> getElasticIpsAsString() throws ServiceException {
+    List<String> addrs = settings.getElasticIps();
+    
+    for(String addr : addrs) {
+      // Validate the ip address pulled from the variables
+      if (!Ip.validIp(addr)) {
+        try {
+          InetAddress.getByName(addr);
+        } catch (UnknownHostException ex) {
+          throw new ServiceException(RESTCodes.ServiceErrorCode.ELASTIC_SERVER_NOT_AVAILABLE,
+            Level.SEVERE, null, ex.getMessage(), ex);
+        }
+      }
+    }
+    return addrs;
+  }
+  
+  private void processException(ServiceException ex) {
```

Comment:
@o-alex have we decided? :)

Reply:
man this was suggested as a elasticclient refactor - start ... it is by no means complete... i did not go through all exceptions and figure out if one is specific to this particular query, or if maybe it is a timeout ...or a elastic is overloaded exception...or something similar

Reply:
@o-alex I was referring to the comment ` //decide if exception requires restart of client`. We can add a TODO or remove the comment. Most probably someone that has never seen this code will need to maintain it and they wouldn't know what to do

Reply:
We probably do not need this method since all exceptions should be caught and logged here https://github.com/logicalclocks/hopsworks/blob/master/hopsworks-api/src/main/java/io/hops/hopsworks/api/exception/mapper/RESTApiThrowableMapper.java